login/service.go

```diff
@@ -162,18 +161,29 @@ func (keycloak *KeycloakOAuthProvider) Perform(ctx *app.AuthorizeLoginContext, a
 	}
 
 	// First time access, redirect to oauth provider
+	redirect := ctx.Redirect
+	referrer := ctx.RequestData.Header.Get("Referer")
```

Comment:
Shouldn't there be some kinda referer whitelist?

Else any website in the world could redirect a user here and get redirected back with a valid dsaas token and hijeck the account..